core/frontend/src/tile/BatchedTileIdMap.ts

```diff
@@ -24,20 +24,25 @@ export class BatchedTileIdMap implements BatchTableProperties {
     this._iModel = iModel;
   }
 
-  /** Obtains or allocates the Id64String corresponding to the supplied set of JSON properties. */
-  public getBatchId(properties: any): Id64String {
+  /** Obtains or allocates the Id64String corresponding to the supplied set of JSON properties.
+   * If allowDuplicates is true, entries with the same properties will be allowed to have different Id64Strings, default is false.
+  */
+  public getBatchId(properties: any, allowDuplicates?: boolean): Id64String {
```

Comment:
My reading of the spec suggests there is no concept of the same "feature" existing within multiple tiles (e.g., crossing tile boundaries or LODs) - you will just have multiple features in each tile that happen to have all the same property values (and may or may not have one or more property values serving as a unique identifier across tiles). So, when the user clicks on a single feature in a tile, that might hilite only a portion of the "real" feature, and the object will no longer be hilited if the user zooms in/out causing that specific tile to fail the LOD test.

Did I overlook something? If not, we're eventually going to have to solve that problem (ideally by amending the spec and/or adding an extension).

Reply:
Correct, selecting a single element by clicking on it will no longer work as you zoom in or out, this information is just not present in the tileset, and even it is present we would need a non-gltf-standard way to assign it as element id, unless we try to match elements with identical properties across tiles, with using the default allowDuplicates=false. Maybe it's not a huge problem to "merge" two instances after all as they are very likely located next to each other anyway, and selection on the UI side will always be driven by their properties.

Reply:
Added element id matching for instances with similar properties